src/cargo/ops/cargo_install.rs

```diff
@@ -189,7 +189,11 @@ fn install_one(
             krate,
             vers,
             config,
-            is_first_install,
+            if is_first_install {
+                NeedsUpdate::TryWithoutFirst
+            } else {
+                NeedsUpdate::False
+            },
```

Comment:
This logic feels not quite right perhaps? If you're installing multiple crates and don't need to update because of the first crate you might still need to update because of the second crate I believe?